dbms/src/Flash/Mpp/GRPCReceiverContext.cpp

```diff
@@ -335,26 +275,28 @@ bool GRPCReceiverContext::supportAsync(const ExchangeRecvRequest & request) cons
     return enable_async_grpc && !request.is_local;
 }
 
+void GRPCReceiverContext::establishMPPConnectionLocal(
+    const ExchangeRecvRequest & request,
+    size_t source_index,
+    const String & req_info,
+    LocalRequestHandler & local_request_handler,
+    bool is_fine_grained)
+{
+    RUNTIME_CHECK_MSG(request.is_local, "This should be a local request");
+
+    auto [tunnel, err_msg] = task_manager->findTunnelWithTimeout(request.req.get(), std::chrono::seconds(10));
+    checkLocalTunnel(tunnel, err_msg);
+    tunnel->connectLocal(source_index, req_info, local_request_handler, is_fine_grained);
+}
+
 ExchangePacketReaderPtr GRPCReceiverContext::makeReader(const ExchangeRecvRequest & request) const
```

Comment:
Maybe can rename it to `makeSyncReader`?

Reply:
> Maybe can rename it to `makeSyncReader`?

Done